src/lib/datatypes/sol-str-slice.c

```diff
@@ -113,3 +115,40 @@ sol_str_slice_split(const struct sol_str_slice slice,
     sol_vector_clear(&v);
     return v;
 }
+
+SOL_API bool
+sol_str_slice_split_iterate(const struct sol_str_slice slice,
+    struct sol_str_slice *token, const char **itr, const char *delim)
```

Comment:
it would be better to make this a slice, and in the header you do one helper that converts str to slice. That way if delim is constant, the compiler will optimize the strlen and it will be slightly faster.